flutter-idea/src/io/flutter/run/daemon/DevToolsServerTask.java

```diff
@@ -0,0 +1,399 @@
+      final boolean dartDevToolsSupported = dartSdkSupportsDartDevTools();
+      if (!dartDevToolsSupported) {
+        progressIndicator.setFraction(60);
+        progressIndicator.setText2("Daemon set-up");
+        setUpWithDaemon();
+        return;
+      }
+
+      // If we are in a Bazel workspace, start the server.
+      // Note: This is only for internal usages.
+      final WorkspaceCache workspaceCache = WorkspaceCache.getInstance(project);
+      if (workspaceCache.isBazel()) {
+        progressIndicator.setFraction(60);
+        progressIndicator.setText2("Starting server");
+        setUpWithDart(createCommand(workspaceCache.get().getRoot().getPath(), workspaceCache.get().getDevToolsScript(),
+                                    ImmutableList.of("--machine")));
+        return;
+      }
+
+      // This is only for development to check integration with a locally run DevTools server.
+      // To enable, follow the instructions in:
+      // https://github.com/flutter/flutter-intellij/blob/master/CONTRIBUTING.md#developing-with-local-devtools
+      final String localDevToolsDir = Registry.stringValue(LOCAL_DEVTOOLS_DIR);
+      if (!localDevToolsDir.isEmpty()) {
+        progressIndicator.setFraction(60);
+        progressIndicator.setText2("Starting local server");
+        setUpLocalServer(localDevToolsDir);
+      }
+
+      // If the Dart plugin does not start DevTools, then call `dart devtools` to start the server.
+      final Boolean dartPluginStartsDevTools = true;
+      if (!dartPluginStartsDevTools) {
+        progressIndicator.setFraction(60);
+        progressIndicator.setText2("Starting server");
+        setUpWithDart(createCommand(DartSdk.getDartSdk(project).getHomePath(),
+                                    DartSdk.getDartSdk(project).getHomePath() + File.separatorChar + "bin" + File.separatorChar + "dart",
+                                    ImmutableList.of("devtools", "--machine")));
+      }
+
+      // Otherwise, wait for the Dart Plugin to start the DevTools server.
+      final CompletableFuture<DevToolsInstance> devToolsFuture = checkForDartPluginInitiatedDevToolsWithRetries(progressIndicator);
+      devToolsFuture.whenComplete((devTools, error) -> {
+        if (error != null) {
+          cancelWithError(new Exception(error));
+        }
+        else {
+          devToolsFutureRef.get().complete(devTools);
+        }
+      });
+    }
+    catch (java.util.concurrent.ExecutionException | InterruptedException e) {
+      cancelWithError(e);
+    }
+  }
+
+  private Boolean dartSdkSupportsDartDevTools() {
+    final DartSdk dartSdk = DartSdk.getDartSdk(project);
+    if (dartSdk != null) {
+      final Version version = Version.parseVersion(dartSdk.getVersion());
+      assert version != null;
+      return version.compareTo(2, 15, 0) >= 0;
+    }
+    return false;
+  }
+
+  @Override
+  public void onThrowable(@NotNull Throwable error) {
+    super.onThrowable(error);
```

Comment:
(Nit.) It looks like this call just logs the error to the `Task.class` logger. Since we're doing our own logging in `cancelWithError`, maybe not necessary to call `super.onThrowable(..)`?

Reply:
Ahh good point, removed